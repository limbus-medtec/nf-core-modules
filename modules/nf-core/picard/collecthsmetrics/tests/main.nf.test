// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test picard/collecthsmetrics
nextflow_process {

    name "Test Process PICARD_COLLECTHSMETRICS"
    script "../main.nf"
    process "PICARD_COLLECTHSMETRICS"

    tag "modules"
    tag "modules_nfcore"
    tag "picard"
    tag "picard/collecthsmetrics"

    test("sarscov2 - bam") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['baits_interval_list'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['targets_interval_list'], checkIfExists: true)
                    ]
                input[1] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)]
                input[2] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)]
                input[3] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['baits_interval_list'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['targets_interval_list'], checkIfExists: true)
                    ]
                input[1] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)]
                input[2] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)]
                input[3] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - bam - nofasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['baits_interval_list'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['targets_interval_list'], checkIfExists: true)
                    ]
                input[1] = [[:],[]]
                input[2] = [[:],[]]
                input[3] = [[:],[]]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - bam - bed") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['baits_bed'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
                    ]

                input[1] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)]
                input[2] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)]
                input[3] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - bam - samebed") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['baits_bed'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['baits_bed'], checkIfExists: true)
                    ]

                input[1] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)]
                input[2] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)]
                input[3] = [[id:'genome'], file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}