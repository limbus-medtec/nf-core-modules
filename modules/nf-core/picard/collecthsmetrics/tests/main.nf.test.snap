{
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-07T10:31:52.003489"
    },
    "sarscov2 - bam - nofasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,1b091a3a08ef3ec5c6c58a9e816614c4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,1b091a3a08ef3ec5c6c58a9e816614c4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-07T10:42:35.220964"
    },
    "sarscov2 - bam - samebed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,c320a9d8088ed4f70f1d85900596472c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,c320a9d8088ed4f70f1d85900596472c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-07T10:43:29.4803"
    },
    "sarscov2 - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,7b2a314db78df3da460f816e2e05a915"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,7b2a314db78df3da460f816e2e05a915"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-07T10:42:05.289673"
    },
    "sarscov2 - bam - bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,847e39b50d8da6707e2ac482d4926706"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.CollectHsMetrics.coverage_metrics:md5,847e39b50d8da6707e2ac482d4926706"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e2a565b586e0aecbb72f1f0d27df7b8d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-06-07T10:43:01.415543"
    }
}